### PR TITLE
Legend Overlay PR.

### PR DESCRIPTION
**Pull request by luciencd**
I added a bit of html+css and angular to add a legend to the visualization that can be opened and closed with a button, and looks nice enough to be in our stakeholder 2 presentation. 

If anyone has stylistic changes to make, I suggest waiting a few days to add that in, in a new branch.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "d3-webpack-loader": "^0.1.1",
     "font-awesome": "^4.7.0",
     "jquery": "^3.2.1",
+    "leaflet-legend": "^1.0.2",
```
Comment (SaswatB) [inline client/package.json]: Remove unused dependency

File: client/src/app/utils/graph.js [hunk 2]
```diff
@@ -7,7 +7,12 @@ class Graph {
         this.width = 1200;
         this.height = 900;
         this.svg = d3.select(element).append("svg");
+
         this.svg.attr("width", this.width).attr("height", this.height).style("background-color", "white");
+
+
+
+
```
Comment (SaswatB) [inline client/src/app/utils/graph.js]: Why??>???????

Comment (luciencd) [inline client/src/app/utils/graph.js]: no idea.

File: client/package.json (hunk 1 above)
Comment (luciencd) [inline client/package.json]: changed

File: client/src/app/utils/graph.js (hunk 2 above)
Comment (luciencd) [inline client/src/app/utils/graph.js]: changed

**Comment by SaswatB** [comment]
Normally inline styles are not allowed, but we'll fix it later
Looks good to me

**Comment by reitakCS** [comment]
Tested and looks good

**Pull request merged**